### PR TITLE
Rename Types to TypeDefinitionSet

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

For historical reasons, we had a disconnect where our set of `TypeDefinitions` was called simply `Types`. This can be confusing when we also have a separate `TypeSet`.

This is primarily a large set of mechanical changes:

* In `astmodel`, rename `Types` to `TypeDefinitions`
* Rename parameters called `types` to `definitions`
* Rename members with names suffixed `types` to use suffix `definitions`
* Rename functions `Find*Types()` to `Find*Definitions()`
* Manual review of comments affected by above refactorings

We'll likely have to follow this up with some manual changes as we clean up the remaining inconsistencies.

**How does this PR make you feel**:
gif

**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/tools/generator/internal/functions/property_assignment_function_test.go [hunk 1]
```diff
@@ -141,7 +141,7 @@ func CreatePropertyAssignmentFunctionTestCases() []*StorageConversionPropertyTes
 			astmodel.MakeTypeName(vNext, "Person"),
 			hubType)
 
-		types := make(astmodel.Types)
+		types := make(astmodel.TypeDefinitionSet)
```
Comment (matthchr) [inline v2/tools/generator/internal/functions/property_assignment_function_test.go:144]: minor: defs (and below)?

**Review by matthchr — COMMENTED** [review]
Looks mostly good. Main comment/question is on the user facing documentation

File: v2/api/authorization/versions_matrix.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-| Types defined in package "authorization"    | v1alpha1api20200801preview |
+| Type Definitions in package "authorization" | v1alpha1api20200801preview |
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/api/authorization/versions_matrix.md:1]: Ah, doh I just realized that these files aren't consumed by the user that much. So actually this isn't that big of a deal either way.

I still think the above is probably true but it doesn't feel nearly as important. For some reason I thought this was the main landing page that was also hosted on our website.

File: v2/tools/generator/internal/astmodel/allof_type.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func BuildAllOfType(types ...Type) Type {
 	return &AllOfType{uniqueTypes}
 }
 
-// Types returns what types the AllOf can be.
+// TypeDefinitionSet returns what types the AllOf can be.
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/allof_type.go:92]: Fixed.

File: v2/tools/generator/internal/astmodel/renaming_visitor.go [hunk 4]
```diff
@@ -64,12 +64,12 @@ func (r *RenamingVisitor) Rename(t Type) (Type, error) {
 	return r.visitor.Visit(t, nil)
 }
 
-// RenameAll applies the renames to the Types
-func (r *RenamingVisitor) RenameAll(types Types) (Types, error) {
-	result := make(Types)
+// RenameAll applies the renames to the definitions
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/renaming_visitor.go:67]: Fixed.

File: v2/tools/generator/internal/astmodel/type_walker.go [hunk 5]
```diff
@@ -38,14 +38,14 @@ type TypeWalker struct {
...
-	result     Types
+	result     TypeDefinitionSet
 	processing map[TypeName]struct{}
 }
 
 // NewTypeWalker returns a TypeWalker.
 // The provided visitor visitTypeName function must return a TypeName and visitObjectType must return an ObjectType or calls to
 // Walk will panic.
-func NewTypeWalker(allTypes Types, visitor TypeVisitor) *TypeWalker {
+func NewTypeWalker(allTypes TypeDefinitionSet, visitor TypeVisitor) *TypeWalker {
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/astmodel/type_walker.go:48]: Fixed.

File: v2/tools/generator/internal/codegen/embeddedresources/remover.go [hunk 6]
```diff
@@ -51,33 +51,33 @@ func (e resourceRemovalVisitorContext) WithMoreDepth() resourceRemovalVisitorCon
...
 
-	resourcePropertiesTypes, err := findAllResourcePropertiesTypes(types)
+	resourcePropertiesTypes, err := findAllResourcePropertiesTypes(definitions)
 	if err != nil {
 		return EmbeddedResourceRemover{}, errors.Wrap(err, "couldn't find resource \"Properties\" type names")
 	}
 
 	remover := EmbeddedResourceRemover{
-		types:                    types,
+		types:                    definitions,
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/embeddedresources/remover.go:80]: Done.

File: v2/tools/generator/internal/codegen/pipeline/state.go [hunk 7]
```diff
@@ -12,8 +12,8 @@ import (
 
 // State is an immutable instance that captures the information being passed along the pipeline
 type State struct {
-	types           astmodel.Types           // set of types generated so far
-	conversionGraph *storage.ConversionGraph // graph of transitions between packages in our conversion graph
+	types           astmodel.TypeDefinitionSet // set of types generated so far
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/state.go:15]: Will do in a followup PR.

File: v2/tools/generator/internal/codegen/pipeline/state.go [hunk 8]
```diff
@@ -38,9 +38,9 @@ func NewState(typeses ...astmodel.Types) *State {
 }
 
 // WithTypes returns a new independentState with the given types instead
-func (s *State) WithTypes(types astmodel.Types) *State {
+func (s *State) WithTypes(definitions astmodel.TypeDefinitionSet) *State {
```
[the remark this answers is not in the document]
Comment (theunrepentantgeek) [reply v2/tools/generator/internal/codegen/pipeline/state.go:41]: Will do in a followup PR.

**Review by matthchr — APPROVED** [review]

**Pull request merged**